### DISCUSSION TITLE
column specific groupByFn

### DISCUSSION THREAD

**Discussion by sgratzl** [TanStack/table#2675] [General]
atm one can only specify one groupByFn for the whole instance. However, I'd like to implement custom grouping functions for different column (types). E.g., a custom one for a numeric column which bins the values in multiple groups. Howe can I do that?

**Comment by gargroh** [comment]
Currently this is not possible, but will be easy to add this option.

**Comment by Kamino0** [comment]
@sgratzl, did you manage to somehow accomplish this?

**Comment by sgratzl** [comment]
atm I'm using the following group by function:

```ts
import { Row, IdType, defaultGroupByFn } from 'react-table';

export function columnSpecificGroupByFn<D extends object = {}>(
  rows: Row<D>[],
  columnId: IdType<D>
): Record<string, Row<D>[]> {
  if (rows.length === 0) {
    return {};
  }
  const column = rows[0].allCells.find((d) => d.column.id === columnId);
  if (column && typeof (column.column as any).groupBy) {
    return (column.column as any).groupBy(rows, column.column);
  }
  return defaultGroupByFn(rows, columnId);
}
```
